File: an2_sem4/MPP/Homework/sapt78/cs/FestivalSellpoint/FestivalSellpoint.Service/IService.cs
using FestivalSellpoint.Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FestivalSellpoint.Service
{
    public interface IService<ID, E> where E : Entity<ID>
    {
        void Add(E e);
        void Update(E e);
        void Remove(ID id);
        E GetById(ID id);
        IEnumerable<E> GetAll();
    }
}

File: an2_sem4/MPP/Homework/sapt9/java/service/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/service/IService.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.service;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Entity;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.EntityRepoException;

public interface IService<ID, E extends Entity<ID>> {
    void add(E e) throws EntityRepoException;
    void update(E e) throws EntityRepoException;
    void remove(ID id) throws EntityRepoException;
    E getById(ID id) throws EntityRepoException;
    Iterable<E> getAll() throws EntityRepoException;
}
